### DISCUSSION TITLE
How do I release the lock for script files to reload them?

### DISCUSSION THREAD

**Discussion by timlindnet** [johnlindquist/kit#1228] [Q&A]
I am syncing scripts from a remote server, all working fine to replace the local ones, but after it I get:
`EBUSY: resource busy or locked, open`

First I clean the scripts folder, then I pull the new files, one await exec() for each
It works, but I find it that it would be better to have a way to cleanly tell kit to unload a set of kenv-scripts on the run before the sync and then load them after. Any idea for this?

I've tried the hide-examples version (altering KIT_EXCLUDE_KENVS w. writeFile run, main..) with no luck.

**Comment by johnlindquist** [comment]
@timlindnet Are you on mac/windows/linux?

I don't think Kit would be locking them for any reason. Kit uses "chokidar" to watch the scripts directory, but that's all.

Any chance they're open in an editor that might be locking them?

**Reply by timlindnet** [reply]
Windows.
I see. VSCode is off but I still get the error. Some seem to experience a similar error connected to chokidar, but I believe in my case the replacing of files is what might confuse it, not sure how to solve that though. A way to unwatch+startwatch during runtime perhaps, or some miss from my side in the script.

Error ::
Kit 1.54.51Electron 24.1.2win32 10.0.19044Locale: en-US
Error: EBUSY: resource busy or locked, open '{file_path}'
::

The script so far:
`/*
## Sync scripts to a linux-server thru ssh with scp
*/

// Name: sync - Sync scripts to server
// Shortcode: sync
// Author: TimLind.net

import "@johnlindquist/kit";
import { join, dirname } from "path";
import { fileURLToPath } from "url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const localDir = join(__dirname, "../scripts");
const remote = `{USER}@{HOST_NAME}`;
const remoteDir = `/some_path/scripts`;
const localSSHKey = `~/.ssh/id_ed25519`;

// OS-specifics
const deleteCommand = (isWin && "del /q") || (isLinux && "rm"); // mac: ?

const removeRemoteFiles = `ssh -i ${localSSHKey} ${remote} rm -f ${remoteDir}/*`;
const removeLocalFiles = `${deleteCommand} ${join(localDir, "*")}`;
const saveFiles = `scp -i ${localSSHKey} -r ${localDir}/* ${remote}:${remoteDir}`;
const getFiles = `scp -i ${localSSHKey} -r ${remote}:${remoteDir}/* ${localDir}`;


let choice = await arg("Sync..", ["Save", "Get"]);

if (choice == "Save") {
  await exec(`${removeRemoteFiles}`);
  await exec(`${saveFiles}`);
} else if (choice == "Get") {
  await exec(`${removeLocalFiles}`);
  await exec(`${getFiles}`);
}
`

**Reply by johnlindquist** [reply]
I could add a way to disable all the file watchers.

How do this sound:

Adding this to ~/.kenv/.env:
`KIT_DISABLE_WATCHERS=true`
Would shut off all the watchers.
Then when you're done, you could remove that line.

That way it's both scriptable and configurable.

**Reply by timlindnet** [reply]
Sounds like a solution as good as any to me. Can't say I'm 100% sure it would solve the lock problem, but all I got is the chokidar watcher as a suspect on this one too.

Should I use inspiration from `account.ts` to do the change(removal) in the `.env` perhaps?:
`await replace({
        files: kitPath(".env"),
        from: /KIT_DISABLE_WATCHERS=.*/g,
        to: ``,
        disableGlobs: true,
      })`

**Reply by johnlindquist** [reply]
Looks right, there's some code in "set-env-var" that would do the trick too. On my phone right now though.

I'll try to add the feature to the next build

**Reply by timlindnet** [reply]
Splendid!
